### ISSUE TITLE
Extract Get Firefox Button into a component

### ISSUE THREAD

**Issue by rebmullin** [mozilla/addons-frontend#6901]
Extract the "Get Firefox" button into its own component.

For QA: Please verify that the "Get Firefox" still appears where it is expected to appear and hasn't changed its appearance in any way.

**Comment by willdurand** [comment]
I removed the "component: code quality" label because it is not a code quality issue but a requested follow-up issue.

**Comment by willdurand** [comment]
I removed the "component: code quality" label because it is not a code quality issue but a requested follow-up issue to break a big patch in smaller ones.

**Comment by willdurand** [comment]
> since it's resuable code and this involves basically some refactoring, it seems like it is kind of a "code quality" issue to me, no?

ah, that's quite correct. I'll re-add the label. As I understood the comment in the PR, it was more than a "nice to have", but what you said above is valid. Sorry for the confusion.

 Yet, I believe the feature should be built upon

> also, I added the "qa:not needed" but I am wondering if QA should double check nothing is broken

Yep, we should use `qa: not needed` with parsimony. I am a big fan of QA, even if it means more time for us to write down STRs or instructions about what to check. Here, I think you are right, we can ask QA to verify that nothing is broken. Let's remove the label and add some instructions.

**Comment by bobsilverberg** [comment]
Thanks for the ping @rebmullin. I am actively working on #6603, and at this point I wasn't planning on extracting the Get Firefox button into its own component, but rather having a single component that would return either the Get Firefox button or an install button. But I do think having it as its own component is a good idea. Unless you are already actively working on this I'd be happy to just take it over as it effectively blocks #6603.

Let me know if you've already started on it, or if you'd like me to take it over.

**Comment by rebmullin** [comment]
@bobsilverberg, I haven't started it so if you want to take over that issue, go for it :)

**Comment by bobsilverberg** [comment]
@rebmullin I'm curious, do you really need just a standalone "Get Firefox" button, or is it a case where you're going to display either an install button or a Get Firefox button (like on the Add-on Detail page)? I ask because I'm going to work on a component that encapsulates the logic for the latter via #6603 next, and I'm wondering if you have any special requirements I should know about.

**Comment by bobsilverberg** [comment]
> I made a few comments about this in your PR but basically per our discussion today on the frontend review call, we will need to show different text. I think we also might want to make puffy optional.

I responded to your comments in the PR but for completeness I'll reply here as well. I think we will address this by having different "types" for the `GetFirefoxButton` which will know what the display text is as well as whether the button is `puffy` or not, etc.

**Comment by ioanarusiczki** [comment]
@rebmullin @willdurand 
I verified the buttons on Chrome, Edge (desktop), Chrome and Safari for small devices (for both android and iOS platforms). 
For what I've noticed I filed https://github.com/mozilla/addons-frontend/issues/7020 that looks like reproducible on AMO dev and AMO stage but not on AMO prod. 

- I did not see problems with the Only with Firefox -Get Firefox now. 
- "Download Firefox" is available on all pages .
- Both buttons redirect to https://www.mozilla.org/en-US/firefox/new/?utm_campaign=non-fx-button&utm_content=header-download-button&utm_medium=referral&utm_source=addons.mozilla.org to download FF.
